src/vs/workbench/contrib/preferences/browser/notificationsEditor.ts

```diff
@@ -0,0 +1,521 @@
+
+		this._register(this.notificationList.onContextMenu(e => this.onContextMenu(e)));
+		this._register(this.notificationList.onDidFocus(() => {
+			this.notificationList.getHTMLElement().classList.add('focused');
+		}));
+	}
+
+	private layoutNotificationsList(): void {
+		if (!this.dimension) {
+			return;
+		}
+		let width = this.dimension.width - 27;
+		for (const columnItem of this.columnItems) {
+			if (columnItem.width && !columnItem.proportion) {
+				width = width - columnItem.width;
+			}
+		}
+		for (const columnItem of this.columnItems) {
+			if (columnItem.proportion && !columnItem.width) {
+				columnItem.width = width * columnItem.proportion;
+			}
+		}
+
+		this.layoutColumns(this.columnItems.map(({ column }) => column));
+		const listHeight = this.dimension.height - (DOM.getDomNodePagePosition(this.headerContainer).height + 12 /*padding*/ + 30 /*list header*/);
+		this.notificationListContainer.style.height = `${listHeight}px`;
+		this.notificationList.layout(listHeight);
+	}
+
+	private getIndexOf(listEntry: IListEntry): number {
+		const index = this.listEntries.indexOf(listEntry);
+		if (index === -1) {
+			for (let i = 0; i < this.listEntries.length; i++) {
+				if (this.listEntries[i].id === listEntry.id) {
+					return i;
+				}
+			}
+		}
+		return index;
+	}
+
+	private selectEntry(entry: INotificationItemEntry | number, focus: boolean = true): void {
+		const index = typeof entry === 'number' ? entry : this.getIndexOf(entry);
+		if (index !== -1) {
+			if (focus) {
+				this.notificationList.getHTMLElement().focus();
+				this.notificationList.setFocus([index]);
+			}
+			this.notificationList.setSelection([index]);
+		}
+	}
+
+	focusNotifications(): void {
+		this.notificationList.getHTMLElement().focus();
+		const currentFocusIndices = this.notificationList.getFocus();
+		this.notificationList.setFocus([currentFocusIndices.length ? currentFocusIndices[0] : 0]);
+	}
+
+	selectNotification(entry: INotificationItemEntry): void {
+		this.selectEntry(entry);
+	}
+
+
+	private onContextMenu(e: IListContextMenuEvent<IListEntry>): void {
+		if (!e.element) {
+			return;
+		}
+
+		if (e.element.templateId === NOTIFICATION_ENTRY_TEMPLATE_ID) {
+			const entry = <INotificationItemEntry>e.element;
+			this.selectEntry(entry);
+		}
+	}
+}
+
+class Delegate implements IListVirtualDelegate<IListEntry> {
+
+	getHeight(element: IListEntry) {
+		return 24;
+	}
+
+	getTemplateId(element: IListEntry) {
+		return element.templateId;
+	}
+}
+
+interface NotificationItemTemplate {
+	parent: HTMLElement;
+	columns: Column[];
+	disposable: IDisposable;
+}
+
+class NotificationItemRenderer implements IListRenderer<INotificationItemEntry, NotificationItemTemplate> {
+
+	get templateId(): string { return NOTIFICATION_ENTRY_TEMPLATE_ID; }
+
+	constructor(
+		private notificationsEditor: NotificationsEditor,
+		private instantiationService: IInstantiationService
+	) {
+
+	}
+
+
+	renderTemplate(parent: HTMLElement): NotificationItemTemplate {
```

Comment:
Right now this isn't getting called - so the elements aren't getting added to the table - and I'm not sure why. There are no errors and NotificationRegistry appears to be working correctly.